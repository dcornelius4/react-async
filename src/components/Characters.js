import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

function Characters({ characters }) {
  const characterList = characters.map((character, i) => (
    <li key={i}>
      <Character character={character} />
    </li>
  ));

  return (
    <ul>
      {characterList}
    </ul>
  );
}

Character.propTypes = {
  characters: PropTypes.array.isRequired
};

export default Characters;
