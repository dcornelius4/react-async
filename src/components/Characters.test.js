import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters';

describe('Characters component', () => {
  it('renders a Characters', () => {
    const characters = [
      {
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Human',
        image: '/image'
      },
      {
        name: 'Beth Smith',
        status: 'Alive',
        species: 'Human',
        image: '/image'
      }
    ];
    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
