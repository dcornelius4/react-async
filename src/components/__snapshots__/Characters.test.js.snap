// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters component renders a Characters 1`] = `
<ul>
  <li
    key="0"
  >
    <Character
      character={
        Object {
          "image": "/image",
          "name": "Rick Sanchez",
          "species": "Human",
          "status": "Alive",
        }
      }
    />
  </li>
  <li
    key="1"
  >
    <Character
      character={
        Object {
          "image": "/image",
          "name": "Beth Smith",
          "species": "Human",
          "status": "Alive",
        }
      }
    />
  </li>
</ul>
`;
